les/fetcher.go

```diff
@@ -412,81 +456,133 @@ func (f *lightFetcher) nextRequest() (*distReq, uint64) {
 			}
 		}
 	}
-	if bestTd == f.maxConfirmedTd {
-		return nil, 0
+	return
+}
+
+// findBestValuesForULC retrieves the best values for ULC mode.
+func (f *lightFetcher) findBestValuesForULC() (bestHash common.Hash, bestAmount uint64, bestTd *big.Int, bestSyncing bool) {
+	bestTd = f.getMaxConfirmedTd()
+	bestSyncing = false
+
+	f.peersLock.RLock()
+	defer f.peersLock.RUnlock()
+	for p, fp := range f.peers {
+		for hash, n := range fp.nodeByHash {
+			if !f.pm.ulc.isTrusted(p.ID()) {
```

Comment:
Do not skip untrusted peers here, you can definitely request headers from them once you trust their hash. It's isTrustedHash where you should skip untrusted peers.